staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -165,17 +165,9 @@ func (f *structuredMergeManager) Apply(liveObj, patchObj runtime.Object, managed
 }
 
 func (f *structuredMergeManager) toVersioned(obj runtime.Object) (runtime.Object, error) {
-	// QUESTION(nodo): That seems reasonable I think, but not clear if this is the right fix yet
-	if obj.GetObjectKind().GroupVersionKind().GroupVersion() == f.groupVersion {
-		return obj, nil
-	}
 	return f.objectConverter.ConvertToVersion(obj, f.groupVersion)
 }
 
 func (f *structuredMergeManager) toUnversioned(obj runtime.Object) (runtime.Object, error) {
-	// QUESTION(nodo): That seems reasonable I think, but not clear if this is the right fix yet
-	if obj.GetObjectKind().GroupVersionKind().GroupVersion() == f.groupVersion {
```

Comment:
would be good to squash the fix into this so we don't have an intermediate commit with this... this is really confusing to see in the commit history

Reply:
Sounds good.